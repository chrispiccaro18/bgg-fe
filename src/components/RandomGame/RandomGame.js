import React, { useState, useEffect } from 'react';
import { getCollection } from '../../services/game-collection-api';
import {
  randomBGPrompt,
  randomBGButton,
  randomBGLabel,
} from '../../utils/strings';

const RandomGame = () => {
  const [username, setUsername] = useState('');
  const [submittedUsername, setSubmittedUsername] = useState('');
  const [randomGame, setRandomGame] = useState('');

  const handleChange = event => {
    setUsername(event.target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();
    setSubmittedUsername(username);
  };

  useEffect(() => {
    if(submittedUsername) {
      getCollection(submittedUsername)
        .then(game => setRandomGame(game));
    }
  }, [submittedUsername]);

  return (
    <>
    <h1>{randomBGPrompt}</h1>
      <form onSubmit={handleSubmit}>
        <label>
          {randomBGLabel}
          <input value={username} onChange={handleChange} />
        </label>
        <button>{randomBGButton}</button>
      </form>
      {randomGame && <h2>{randomGame.name}</h2>}
    </>
  );
};

export default RandomGame;
