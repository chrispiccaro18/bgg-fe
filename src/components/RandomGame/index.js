import RandomGame from './RandomGame';

export default RandomGame;
