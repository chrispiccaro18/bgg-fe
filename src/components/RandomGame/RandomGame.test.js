import React from 'react';
import { shallow } from 'enzyme';
import RandomGame from './RandomGame';

describe('RandomGame component', () => {
  it('renders RandomGame', () => {
    const wrapper = shallow(<RandomGame />);
    expect(wrapper).toMatchSnapshot();
  });
});
  