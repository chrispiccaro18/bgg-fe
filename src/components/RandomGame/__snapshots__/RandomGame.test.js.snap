// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomGame component renders RandomGame 1`] = `
<Fragment>
  <h1>
    Enter Your BoardGameGeek Username to get a random game from your collection.
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
      BoardGameGeek Username:
      <input
        onChange={[Function]}
        value=""
      />
    </label>
    <button>
      Random Game
    </button>
  </form>
</Fragment>
`;
